activerecord/lib/active_record/relation/predicate_builder/range_handler.rb

```diff
@@ -14,17 +14,25 @@ def initialize(predicate_builder)
       end
 
       def call(attribute, value)
-        begin_bind = predicate_builder.build_bind_attribute(attribute.name, value.begin)
-        end_bind = predicate_builder.build_bind_attribute(attribute.name, value.end)
-        if value.begin.respond_to?(:infinite?) && value.begin.infinite?
-          if value.end.respond_to?(:infinite?) && value.end.infinite?
-            attribute.not_in([])
+        if !infinity?(value.begin) && !infinity?(value.end) && value.begin > value.end
```

Comment:
This really feels like it's taking more effort to work around the problem than giving the attribute responsibility for ranges.

My examples are getting more obscure, and the gaps are getting smaller... but the fact there are *any* gaps highlights that it feels like a piecemeal solution, with lots of conditionals, instead of few wide-reaching rules.

```ruby
x = Author.create! name: "12 Monkeys"
assert_equal [x], Author.where(name: 10..2)
```

Maybe that's silly enough that we don't care.. but it is a change in behaviour. 

(Or maybe I'm misreading and the value's already been cast before this? But I think that only happens below.)

I'll defer to @sgrif, but I still feel like we want something like https://github.com/rails/rails/pull/30000#discussion_r162247130 instead of having this class expect more from all the value types it might encounter.